test/integration/ipamperf/cloud.go

```diff
@@ -0,0 +1,151 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package ipamperf
+
+import (
+	"encoding/json"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"net"
+	"net/http"
+	"regexp"
+	"sync"
+
+	compute "google.golang.org/api/compute/v1"
+	"google.golang.org/api/googleapi"
+	"k8s.io/kubernetes/pkg/controller/nodeipam/ipam/cidrset"
+	"k8s.io/kubernetes/test/integration/util"
+)
+
+var (
+	pathInstance  = regexp.MustCompile(`/compute/(v[\d]+|beta|alpha)/projects/([\w|-]+)/zones/([\w|-]+)/instances/([\w|-]+)(\?.*)?`)
+	pathUpdateNet = regexp.MustCompile(`/compute/(v[\d]+|beta|alpha)/projects/([\w|-]+)/zones/([\w|-]+)/instances/([\w|-]+)/updateNetworkInterface(\?.*)?`)
+	pathOperation = regexp.MustCompile(`/compute/(v[\d]+|beta|alpha)/projects/([\w|-]+)/zones/([\w|-]+)/operations/([\w|-]+)(\?.*)?`)
+)
+
+func writeAsJSON(w io.Writer, val interface{}) {
+	j, _ := json.Marshal(val)
+	fmt.Fprintf(w, string(j))
+}
+
+func writeOperationStatus(w io.Writer, selfLink string, header http.Header, statusCode int, status string) {
+	writeAsJSON(w, &compute.Operation{
+		ServerResponse: googleapi.ServerResponse{
+			Header:         header,
+			HTTPStatusCode: statusCode,
+		},
+		SelfLink: selfLink,
+		Status:   status,
+	})
+}
+
+type cloudServer struct {
+	qps          int                          // the qps to server at
+	allocateCIDR bool                         // whether cloud should allocate CIDR
+	cidrSet      *cidrset.CidrSet             // the cidr set if allocateCIDR is true
+	instances    map[string]*compute.Instance // known instances
+	lock         sync.Mutex                   // synchronize changes
```

Comment:
Conventionally, we place this lock object above the set of variables it's used for locking. For eg, see - https://github.com/kubernetes/kubernetes/blob/master/pkg/controller/nodeipam/ipam/cloud_cidr_allocator.go#L69

Reply:
Also, you probably can get rid of the comment 'synchronize changes' :)